src/cisco_gnmi/client.py

```diff
@@ -334,3 +340,203 @@ def parse_xpath_to_gnmi_path(self, xpath, origin=None):
+        if len(curr_diff) > 1:
+            print('combine_configs() error1')
+            return payload
+        index = curr_diff.pop()
+        curr_xpath = xpath[xpath.find(index):]
+        curr_xpath = curr_xpath.split('/')
+        curr_xpath.reverse()
+        for seg in curr_xpath:
+            config = {seg: config}
+
+        last_diff = last_set - set(xpath.split('/'))
+        if len(last_diff) > 1:
+            print('combine_configs() error2')
+            return payload
+        last_xpath = last_xpath[last_xpath.find(last_diff.pop()):]
+        last_xpath = last_xpath.split('/')
+        last_xpath.reverse()
+        for seg in last_xpath:
+            if seg not in payload:
+                payload = {seg: payload}
+        payload.update(config)
+        return payload
+
+    def xpath_to_json(self, configs, last_xpath='', payload={}):
+        """Try to combine Xpaths/values into a common payload (recursive).
+
+        Parameters
+        ----------
+        configs: tuple of xpath/value dict
+        last_xpath: str of last xpath that was recusivly processed.
+        payload: dict being recursively built for JSON transformation.
+
+        Returns
+        -------
+        dict of combined xpath/value dict.
+        """
+        for i, cfg in enumerate(configs, 1):
+            xpath, config, is_key = cfg
+            if last_xpath and xpath not in last_xpath:
+                # Branched config here
+                #   |---last xpath config
+                # --|
+                #   |---this xpath config
+                payload = self.combine_configs(payload, last_xpath, xpath, config)
+                return self.xpath_to_json(configs[i:], xpath, payload)
+            xpath_segs = xpath.split('/')
+            xpath_segs.reverse()
+            for seg in xpath_segs:
+                if not seg:
+                    continue
+                if payload:
+                    if is_key:
+                        if seg in payload:
+                            if isinstance(payload[seg], list):
+                                payload[seg].append(config)
+                            elif isinstance(payload[seg], dict):
+                                payload[seg].update(config)
+                        else:
+                            payload.update(config)
+                            payload = {seg: [payload]}
+                    else:
+                        config.update(payload)
+                        payload = {seg: config}
+                    return self.xpath_to_json(configs[i:], xpath, payload)
+                else:
+                    if is_key:
+                        payload = {seg: [config]}
+                    else:
+                        payload = {seg: config}
+                    return self.xpath_to_json(configs[i:], xpath, payload)
+        return payload
+
+    # Pattern to detect keys in an xpath
+    RE_FIND_KEYS = re.compile(r'\[.*?\]')
+
+    def get_payload(self, configs):
```

Comment:
From peer review: Please provide usage example.

Reply:
Thank you, we will want to remove before merge but useful for now.

Reply:
Moved this to new xpath_util.py